language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java

```diff
@@ -676,4 +679,74 @@ protected boolean onSuggestionsAfterQualifiers(BallerinaCompletionContext contex
         Token lastQualifier = qualifiers.get(qualifiers.size() - 1);
         return lastQualifier.textRange().endOffset() < cursor;
     }
+
+    /**
+     * Get completions for anonymous function definition.
+     *
+     * @param context completion context.
+     * @return completion item
+     */
+    protected List<LSCompletionItem> getAnonFunctionDefCompletions(BallerinaCompletionContext context) {
+        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        Optional<TypeSymbol> typeSymbolAtCursor = context.getContextType();
+
+        if (typeSymbolAtCursor.isEmpty()) {
+            return Collections.emptyList();
+        }
+        
+        TypeSymbol symbol = typeSymbolAtCursor.get();
+        if (symbol.typeKind() == TypeDescKind.FUNCTION) {
+            FunctionTypeSymbol functionTypeSymbol = ((FunctionTypeSymbol) symbol);
+            List<String> args = new ArrayList<>();
+            int argIndex = 1;
+
+            Set<String> visibleSymbolNames = visibleSymbols
+                    .stream()
+                    .map(Symbol::getName)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .collect(Collectors.toSet());
+                
+            if (functionTypeSymbol.params().isEmpty()) {
```

Comment:
Shall we check the case where we don't have any parameters? For example, 
```
function myFunc returns function  {
   <cursor>
}
```

Reply:
Generated variable names only if `functionTypeSymbol.params().isPresent()` to support this.